### PR TITLE
feat: Add detection for Notion SAML SSO configuration change

### PR DESCRIPTION
**Pull request by le4ker**
### Background

Detection for SAML SSO configuration changes. 
Fires an `HIGH` severity alert in case the configuration is disabled, otherwise `INFO`

### Changes

* Adds `notion_workspace_settings_enforce_saml_sso_config_updated.py` `notion_workspace_settings_enforce_saml_sso_config_updated.yaml` and udpates the Notion pack.

### Testing

```
Notion.SAML.SSO.Configuration.Changed
        [PASS] Other Event
                [PASS] [rule] false
        [PASS] SAML SSO Enabled
                [PASS] [rule] true
                [PASS] [title] Notion User [homer.simpson@yourcompany.io] updated settings to enable SAML SSO config for workspace id ..
                [PASS] [dedup] Notion User [homer.simpson@yourcompany.io] updated settings to enable SAML SSO config for workspace id ..
                [PASS] [alertContext] {"actor": {"id": "..", "object": "user", "type": "person", "person": {"email": "homer.simpson@yourcompany.io"}}, "action": "workspace.settings.enforce_saml_sso_config_updated"}
                [PASS] [severity] INFO
        [PASS] SAML SSO Disabled
                [PASS] [rule] true
                [PASS] [title] Notion User [homer.simpson@yourcompany.io] updated settings to disable SAML SSO config for workspace id ..
                [PASS] [dedup] Notion User [homer.simpson@yourcompany.io] updated settings to disable SAML SSO config for workspace id ..
                [PASS] [alertContext] {"actor": {"id": "..", "object": "user", "type": "person", "person": {"email": "homer.simpson@yourcompany.io"}}, "action": "workspace.settings.enforce_saml_sso_config_updated"}
                [PASS] [severity] HIGH

```

### CODE REVIEW

File: rules/notion_rules/notion_workspace_settings_enforce_saml_sso_config_updated.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        event,
+        "workspace.settings.enforce_saml_sso_config_updated",
+        "state",
+        default="<NO_STATE_FOUND>",
+    )
+
+    if state == "enabled":
+        return "INFO"
+
+    return "HIGH"
```
Comment (le4ker) [inline rules/notion_rules/notion_workspace_settings_enforce_saml_sso_config_updated.py:48]: if `state` has an unexpected value, better to be  on the conservative (`HIGH`) side

**Review by andrea-youwakim — DISMISSED** [review]
beautiful pr, minor yaml changes before merging

please run the tests for this rule id and paste the output in the pr description for ease of functionality assessment for everyone viewing the pr

File: rules/notion_rules/notion_workspace_settings_enforce_saml_sso_config_updated.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
+AnalysisType: rule
+Description: A Notion User changed settings to enforce SAML SSO configurations for your organization.
+DisplayName: "Notion SAML SSO Configuration Changed"
+Enabled: true
+Filename: notion_workspace_settings_enforce_saml_sso_config_updated.py
+Runbook: Possible Data Exfiltration. Follow up with the Notion User to determine if this was done for a valid business reason.
+Severity: High
+Tags:
+  - Data Security:Data Exfiltration
```
[the remark this answers is not in the document]
Comment (le4ker) [reply rules/notion_rules/notion_workspace_settings_enforce_saml_sso_config_updated.yml]: Good catch, I removed it 😄

**Review by edyesed — APPROVED** [review]
Adds a detection for changes in Notion SAML SSO Config changes. `INFO` severity if sso is enabled, `HIGH` severity for all other conditions

**Pull request merged**